File: 백준/Silver/15652. N과 M （4）/N과 M （4）.py
import sys
input = sys.stdin.readline

def f(i,m):
    if i == m:
        print(*tmp)
        return
    else:
        for k in arr:
            tmp[i] = k
            if tmp[i] >= tmp[i-1]:
                f(i+1, m)
            tmp[i] = 0
n, m = map(int,input().split())
arr = [i for i in range(1, n+1)]
tmp = [0]*m
f(0,m)

File: 백준/Silver/15652. N과 M （4）/N과 M （4）.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const f = (depth) => {
  if (depth == m) {
    output += tmp.join(" ");
    output += "\n";
    return;
  } else {
    for (let i = 1; i <= n; i++) {
      if (depth == 0 || tmp[depth - 1] <= i) {
        tmp.push(i);
        f(depth + 1);
        tmp.pop();
      }
    }
  }
};

const [n, m] = input[0].split(" ");
let output = "";
let tmp = [];
f(0);
console.log(output);
